nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutEmail.java

```diff
@@ -459,7 +470,8 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
                 final MimeBodyPart mimeFile = new MimeBodyPart();
                 session.read(flowFile, stream -> {
                     try {
-                        mimeFile.setDataHandler(new DataHandler(new ByteArrayDataSource(stream, "application/octet-stream")));
+                        final String mimeType = flowFile.getAttribute("mime.type");
+                        mimeFile.setDataHandler(new DataHandler(new ByteArrayDataSource(stream, (mimeType == null || mimeType.isEmpty()) ? "application/octet-stream" : mimeType)));
```

Comment:
It would be helpful to declare a separate variable for the resolved MIME Type instead of the inline ternary.

Reply:
done